stickermanager.py

```diff
@@ -0,0 +1,354 @@
+    if animated:
+        file = await bot.upload_file(current_vote['filepath'])
+        mime = 'application/x-tgsticker'
+    else:
+        img = Image.open(current_vote['filepath'])
+        w, h = img.size
+        if w > h:
+            img = img.resize((512, int(h * (512 / w))), Image.ANTIALIAS)
+        else:
+            img = img.resize((int((w * (512 / h))), 512), Image.ANTIALIAS)
+        dat = BytesIO()
+        img.save(dat, format='PNG')
+        file = await bot.upload_file(dat.getvalue())
+        mime = 'image/png'
+    os.remove(current_vote['filepath'])
+    file = InputMediaUploadedDocument(file, mime, [])
+    document = await bot(UploadMediaRequest(InputPeerSelf(), file))
+    document = utils.get_input_document(document)
+    item = InputStickerSetItem(document=document, emoji=current_vote['emoji'])
+    pack: Optional[StickerSet] = None
+    added = False
+    # TODO add support for animated stickers
+    if not sticker_pack:
+        added, pack = await create_sticker_pack(bot, item)
+    if not added:
+        pack = await bot(AddStickerToSetRequest(stickerset=sticker_pack, sticker=item))
+    return pack, utils.get_input_document(pack.documents[-1])
+
+
+def get_template_data() -> dict:
+    def format_votes(cond: Callable[[Number], bool]) -> str:
+        return ', '.join(VOTE_TEMPLATE.format(uid=uid, displayname=displayname, weight=weight)
+                         for uid, (weight, displayname) in current_vote['votes'].items()
+                         if cond(weight))
+
+    return {
+        **current_vote,
+        'votes': '',
+        'yes': format_votes(lambda weight: weight > 0) or 'nobody',
+        'no': format_votes(lambda weight: weight < 0) or 'nobody',
+        'pack_link': f'https://t.me/addstickers/{STICKER_PACK_SHORT_NAME}',
+    }
+
+
+def calculate_scores() -> Scores:
+    yes = 0.0
+    no = 0.0
+    yes_count = 0
+    no_count = 0
+    for vote in current_vote['votes'].values():
+        if vote.weight > 0:
+            yes += vote.weight
+            yes_count += 1
+        else:
+            no -= vote.weight
+            no_count += 1
+    return Scores(fancy_round(yes - no), fancy_round(yes), fancy_round(no), yes_count, no_count)
+
+
+def fancy_round(val: Number) -> Number:
+    if isinstance(val, float) and val.is_integer():
+        return int(val)
+    return round(val, 2)
+
+
+async def init(bot: TelegramClient) -> None:
+    @bot.on(events.NewMessage(pattern='#addsticker(?: (.+))?', chats=ALLOWED_CHATS))
+    async def start_poll(event: Union[events.NewMessage.Event, Message]) -> None:
+        if not event.is_reply:
+            return
+        elif current_vote:
+            await event.reply('There\'s already an ongoing sticker poll.')
+            return
+
+        async with current_vote_lock:
+            await _locked_start_poll(event)
+            save_cache()
+
+    async def _locked_start_poll(event: Union[events.NewMessage.Event, Message]) -> None:
```

Comment:
…this is not what I intended. The following is:

```python
async def locked_start_poll(event: Union[events.NewMessage.Event, Message]) -> None:
    async with current_vote_lock:
        await _start_poll()
        save_cache()

async def _start_poll(event: Union[events.NewMessage.Event, Message]) -> None:
    ...
```

As it stands, the renaming is pointless.